textattack/shared/scripts/run_attack_parallel.py

```diff
@@ -104,28 +109,39 @@ def run(args):
         num_results = 0
         num_failures = 0
         num_successes = 0
-    pbar = tqdm.tqdm(total=num_remaining_examples, smoothing=0)
-    while num_results < num_total_examples:
+    pbar = tqdm.tqdm(total=num_remaining_attacks, smoothing=0)
+    while num_results < num_total_examples and worklist:
         result = out_queue.get(block=True)
 
         if isinstance(result, Exception):
             raise result
+        idx, result = result
         attack_log_manager.log_result(result)
+        worklist.remove(idx)
+
         if (not args.attack_n) or (not isinstance(result, textattack.attack_results.SkippedAttackResult)):
             pbar.update()
             num_results += 1
+
             if type(result) == textattack.attack_results.SuccessfulAttackResult:
                 num_successes += 1
             if type(result) == textattack.attack_results.FailedAttackResult:
                 num_failures += 1
             pbar.set_description('[Succeeded / Failed / Total] {} / {} / {}'.format(num_successes, num_failures, num_results))
         else:
-            label, text = next(dataset)
```

Comment:
I think this is a good change. Can you remove the `._i` indexing from the `TextAttackDataset` altogether, and remove the `__next__` method from there?